classes/SenseiManager.php

```diff
@@ -0,0 +1,313 @@
+                    'sensei_name' => $row['user_name'],
+                ]
+            );
+        }
+        return $application_list;
+    }
+
+    public static function getApplicationsBySensei(int $sensei_id, System $system): array {
+        $application_list = [];
+        $application_result = $system->query("SELECT `student_applications`.`student_id` as 'student_id', `users`.`user_name` FROM `student_applications`
+                       INNER JOIN `users` ON `student_applications`.`student_id` = `users`.`user_id`
+                       WHERE `student_applications`.`sensei_id` = '{$sensei_id}'");
+        while($row = $system->db_fetch($application_result)) {
+            array_push($application_list, (object) [
+                    'student_id' => $row['student_id'],
+                    'student_name' => $row['user_name'],
+                ]
+            );
+        }
+        return $application_list;
+    }
+
+    public static function updateStudentRecruitment(int $sensei_id, $recruitment_message, System $system): bool {
+        $db_modified = false;
+        $system->query("START TRANSACTION;");
+        $system->query("UPDATE `sensei` SET `recruitment_message` = '{$recruitment_message}' WHERE `sensei_id` = '{$sensei_id}'");
+        $system->query("COMMIT;");
+        if ($system->db_last_num_rows > 0) {
+            $db_modified = true;
+        }
+        return $db_modified;
+    }
+
+    public static function updateStudentSettings(int $sensei_id, $student_message, $specialization, System $system): bool {
+        $db_modified = false;
+        $system->query("START TRANSACTION;");
+        $system->query("UPDATE `sensei` SET `student_message` = '{$student_message}', `specialization` = '{$specialization}' WHERE `sensei_id` = '{$sensei_id}'");
+        $system->query("COMMIT;");
+        if ($system->db_last_num_rows > 0) {
+            $db_modified = true;
+        }
+        return $db_modified;
+    }
+
+    public static function removeStudent(int $sensei_id, int $student_id, System $system): bool {
+        $db_modified = false;
+        $system->query("START TRANSACTION;");
+        $student_result = $system->query("SELECT `students` FROM `sensei` WHERE `sensei_id` = '{$sensei_id}'");
+        $result = $system->db_fetch($student_result);
+        $students = json_decode($result['students']);
+        $students = array_diff($students, array($student_id));
+        $students = "[" . implode(', ', $students) . "]";
+        $system->query("UPDATE `sensei` SET `students` = '{$students}' WHERE `sensei_id` = '{$sensei_id}'");
+        $system->query("UPDATE `users` SET `sensei_id` = '0' WHERE `user_id` = '{$student_id}'");
+        $system->query("COMMIT;");
+        if ($system->db_last_num_rows > 0) {
+            $db_modified = true;
+        }
+        return $db_modified;
+    }
+
+    public static function addSensei(int $sensei_id, $specialization, System $system): bool {
```

Comment:
Similarly here, if you're reading this in isolation it's not clear what the "sensei ID" is supposed to be. If we name it `sensei_user_id` or just `user_id` it makes it a little more clear